### DISCUSSION TITLE
A scriptlet to call a function on load

### DISCUSSION THREAD

**Discussion by D4niloMR** [uBlockOrigin/uBlock-issues#3300] [Ideas]
I ask specifically because I think some cookie filters could be simplified to a simple function call in a non-trusted scriptlet. This would not only reduce the complexity of the filters, but also the time it takes to create them. 

For example:

[OneTrust.RejectAll();](https://my.onetrust.com/s/article/UUID-d8291f61-aa31-813a-ef16-3f6dec73d643?language=en_US#UUID-d8291f61-aa31-813a-ef16-3f6dec73d643_section-idm231900113640268)

[Didomi.setUserDisagreeToAll();](https://developers.didomi.io/cmp/web-sdk/reference/api#setuserdisagreetoall)

It could also be of use in some situations not related to cookies.

**Comment by uBlock-user** [comment]
I wrote a scriptlet a long while ago - https://github.com/uBlockO/uBO-Scriptlets/blob/master/scriptlets.js#L211

Does that work for this case ?

**Reply by D4niloMR** [reply]
No, I tried `tvn24.pl##+js(cf, OneTrust.RejectAll, 1000)` on 

`https://tvn24.pl/tvnmeteo/prognoza/co-dalej-z-pogoda-w-wakacje-modele-z-europy-usa-i-rosji-sa-raczej-zgodne-st7985799`

**Reply by uBlock-user** [reply]
I can't test due to geoblock, I have removed the world param so download the [copy](https://ubo-scriptlets.pages.dev/scriptlets.js) again and test.

**Reply by garry-ut99** [reply]
Still doesn't work, neither on `tvn24.pl`, nor on `ryobitools.eu`.

> I can't test due to geoblock

You can try on `ryobitools.eu` or any similiar site from the publicwww link: https://github.com/uBlockOrigin/uAssets/discussions/20789#discussioncomment-9979857

**Reply by uBlock-user** [reply]
`setTimeout(OneTrust.RejectAll, 10);` works, so will need to change the `runAt` value
